src/screens/studentGroup.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react'
+import { useParams } from 'react-router-dom'
+import 'firebase/firestore'
+import { useFirestore, useUser, useFirestoreDocData, useFirestoreCollectionData } from 'reactfire'
+import { Input, Button, Box, Editable, EditablePreview, EditableInput } from '@chakra-ui/react'
+import { IStudentGroup } from 'interfacesAndTypes/interfacesAndTypes'
+import styled from '@emotion/styled'
+
+interface Params {
+  groupId: string
+}
+
+const StudentNameForm = styled.form``
+
+const StudentBox = styled.div`
+  margin: auto;
+  width: 90%;
+  border: 1px solid black;
+  min-height: 100px;
+`
+
+const StudentGroup: React.FC = () => {
+  const [studentInput, setStudentInput] = React.useState('')
+  const [studentGroupNameInput, setStudentGroupNameInput] = React.useState<string>('')
+
+  const params: Params = useParams()
+  const studentGroupId = params.groupId
+
+  const { data: user } = useUser()
+
+  const studentGroupRef = useFirestore()
+    .collection('teachers')
+    .doc(user.uid)
+    .collection('studentGroups')
+    .doc(studentGroupId)
+  const studentGroupDocument = useFirestoreDocData<IStudentGroup & { docId: string }>(studentGroupRef, {
+    idField: 'docId',
+  }).data
+
+  const studentsRef = useFirestore().collection('teachers').doc(user.uid).collection('students')
+  const students = useFirestoreCollectionData(studentsRef, { idField: 'docId' })
+
+  React.useEffect(() => {
+    if (studentGroupDocument) {
+      setStudentGroupNameInput(studentGroupDocument.studentGroupName)
+    }
+  }, [studentGroupDocument])
```

Comment:
I'm not sure this is needed.  Why can't we simply render the data off `studentGroupDocument`?

I think you're trying to turn the Chakra Editable component into a controlled component with a useState hook.  That's more firepower than we need, unless I'm missing something.  Why can't we simply fire off the `update` request to firebase on the `onSubmit` event coming to us from `Editable`?

Reply:
I see what you're saying about the useState, I went ahead and got rid of that.  The problem I'm having is that the studentGroup name isn't showing up in the editable component because it's rendering before the studentGroupDocument is fetched.  This makes it work:

```
{studentGroupDocument ? (
          <Editable
            defaultValue={studentGroupDocument.studentGroupName}
            onSubmit={editStudentGroupNameHandler}
            fontSize="2.5rem"
          >
            <EditablePreview _hover={{ cursor: 'pointer' }} />
            <EditableInput />
          </Editable>
        ) : null}
```
Is there a cleaner solution?

Reply:
i think `{ studentGroupDocument && <Editable> ... }` might work

Reply:
Cool, that works.  pushing all fixes now